### PR TITLE
making calico hosted (and updating to latest)

### PR DESCRIPTION
**Pull request by puja108**
Towards https://github.com/giantswarm/k8scloudconfig/issues/40

WIP

Running Calico 2.2.1

This is already implemented and tested manually on `wopre` cluster on AWS FRA

TODO:
- [x] add calico yamls
- [x] check/rename etcd cert directory/filename
- [ ] test
- [ ] retag images

### CODE REVIEW

**Comment by puja108** [comment]
Open question: Do we still need to set MTU manually?

**Comment by calvix** [comment]
@puja108  for KVM on-prem yes we need

**Comment by puja108** [comment]
@calvix so could we make this optional? I don't feel good adjusting MTU when not needed on AWS

File: templates.go [hunk 1]
```diff
@@ -3,27 +3,320 @@ package cloudconfig
...
+      namespace: kube-system
+    data:
+      # Configure this with the location of your etcd cluster.
+      etcd_endpoints: "https://{{ .Cluster.Etcd.Domain }}:2379"
+
+      # Configure the Calico backend to use.
+      calico_backend: "bird"
+    
+      # The CNI network configuration to install on each node.
+      # TODO: Do we still need to set MTU manually?
```
Comment (rossf7) [inline templates.go]: Not urgent but would be good to know about MTU. As we can also remove from clustertpr if we don't need it.

File: templates.go [hunk 2]
```diff
@@ -3,27 +3,320 @@ package cloudconfig
...
+      selector:
+        matchLabels:
+          k8s-app: calico-node
+      template:
+        metadata:
+          labels:
+            k8s-app: calico-node
+          annotations:
+            scheduler.alpha.kubernetes.io/critical-pod: ''
+            scheduler.alpha.kubernetes.io/tolerations: |
```
Comment (rossf7) [inline templates.go]: This is really nice.

File: templates.go (hunk 1 above)
Comment (puja108) [inline templates.go]: @calvix mentioned we need it in KVM, so I created a ticket in operator board for making it optional

Comment (puja108) [inline templates.go]: https://github.com/giantswarm/k8scloudconfig/issues/126

**Pull request merged**